src/controller.rs

```diff
@@ -144,18 +143,27 @@ impl Controller {
     }
 
     pub fn on_keypress_quit(&self) -> Result<(), Box<dyn Error>> {
-        trace!("on_keypress_quit");
+        info!("on_keypress_quit");
+
+        // first set state flag that indicates to the rest of the app that
+        // we are actively quiting
+        self.lock_and_load(|state| Ok(Some(StateMutation::on(state).begin_quitting().commit())))?;
+
+        // time to massacre all processes
         self.lock_and_load(|state| {
```

Comment:
this is all i can think about when i read this function name
![image](https://github.com/erhickey/proctmux/assets/426949/850b1191-2753-4657-8fff-c20e96c015bb)